src/util/createValidation.js

```diff
@@ -46,6 +50,43 @@ function createErrorFactory({ value, label, resolve, originalValue, ...opts}) {
     , { params })
   }
 }
+export function createOneOfTypeValidation(options) {
```

Comment:
I don't think you need to create a new helper here. test functions have access to the schema instance (via `this`) so if you store the possible types on the schema you can access them in a test and call out to the inner types, without needing to add a new helper here

Reply:
This is what I had first, but sync validation fails with : 
``` 
Error: Validation test of type: "oneOfType" returned a Promise during a synchronous validate. This test will finish after the validate call has returned
```

```
    next._oneOfTypeError = createValidation({
        message,
        test(value) {
          const {sync} = this.options;
//            console.log("sync? ",this);
            return validTypes.reduce((valid, schema)=>{
                if(!schema.validate) return valid;
                return schema.validate(value, {sync})
                             .then(validated=> validated || valid)
                             .catch(()=> valid);
            }, promise(sync).resolve(invalidMarker)).then(result=>{
                if(result === invalidMarker){
                    return this.createError({
                        params:{
                            values: validTypes.map(subType => typeOf(subType)).join(', ')
                        }
                    });
                }
                return result;
            });
        },
```
Any ideas?

Reply:
yeah i think the check that throws that error doesn't check if the promise is a sync one. Maybe we should flag the promise object as synchronous so the error can check and ignore in this case